### PR TITLE
Latest coverage updates and npmignore

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

**Comment by dignifiedquire** [comment]
cc @RichardLitt @diasdavid

**Comment by diasdavid** [comment]
awesome! :D

File: js-project-guidelines.md [hunk 1]
```diff
@@ -179,17 +183,24 @@ Inside the package.json, the main file exported is the one from the auto-generat
...
+  - npm run coverage
+after_success:
+  - npm run coverage-publish
+```
+
 ##### `.gitignore`
 
-To avoid checking in the dist and lib folders, the `.gitignore` file should at least contain:
+To avoid checking in unwanted files, the `.gitignore` file should follow
+this [example](examples/.gitignore).
```
Comment (RichardLitt) [inline js-project-guidelines.md]: The word should is a bit hard. This should be reworded to say 'A .gitignore is useful. An example is provided in examples'

Comment (dignifiedquire) [inline js-project-guidelines.md]: I disagree, not having this `.gitignore` can mess up a lot of the rest in this document, and create a lot of other problems, so having if is a "should" surley

Comment (RichardLitt) [inline js-project-guidelines.md]: This document isn't only for large js-ipfs-* style projects. It is also for small javascript projects that frankly do not need all of these things. The .gitignore here seems excessively bloated for some smaller repos. For example, [dnslink-deploy](https://github.com/ipfs/dnslink-deploy) is a JS repo that doesn't need this.

Comment (diasdavid) [inline js-project-guidelines.md]: Note that the .gitignore you see is pretty much the .gitignore you get when you create a repo on github and select "Node.js" as the type of project.

Comment (RichardLitt) [inline js-project-guidelines.md]: But it also has unnecessary things, like `.grunt`. I feel that it is overkill, and thus we shouldn't use the word _should_. Do you see what I mean? Am I being unreasonable, here?

**Pull request merged**